internal/registry/client.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package registry
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+const (
+	productionURL = "https://epr.elastic.co"
+	stagingURL    = "https://epr-staging.elastic.co"
+	snapshotURL   = "https://epr-snapshot.elastic.co"
+)
+
+var (
+	// Production is a pre-configured production client
+	Production = NewClient(productionURL)
+	// Staging is a pre-configured staging client
+	Staging = NewClient(stagingURL)
+	// Snapshot is a pre-configured snapshot client
+	Snapshot = NewClient(snapshotURL)
+)
+
+// Client is responsible for exporting dashboards from Kibana.
+type Client struct {
+	host string
+}
+
+// NewClient creates a new instance of the client.
+func NewClient(host string) *Client {
+	return &Client{
+		host: host,
+	}
+}
+
+func (c *Client) get(resourcePath string) (int, []byte, error) {
+	return c.sendRequest(http.MethodGet, resourcePath, nil)
+}
+
+func (c *Client) post(resourcePath string, body []byte) (int, []byte, error) {
+	return c.sendRequest(http.MethodPost, resourcePath, body)
+}
+
+func (c *Client) put(resourcePath string, body []byte) (int, []byte, error) {
+	return c.sendRequest(http.MethodPut, resourcePath, body)
+}
+
+func (c *Client) delete(resourcePath string) (int, []byte, error) {
+	return c.sendRequest(http.MethodDelete, resourcePath, nil)
+}
```

Comment:
I understand why you have put all these methods, but as long as they're not used by the client, I would be for deleting them. If you prefer you can merge `get` and `sendRequest` methods (KISS).